x-pack/plugins/security_solution/public/common/components/event_details/event_fields_browser.tsx

```diff
@@ -166,10 +174,22 @@ const useFieldBrowserPagination = () => {
 
 /** Renders a table view or JSON view of the `ECS` `data` */
 export const EventFieldsBrowser = React.memo<Props>(
-  ({ browserFields, data, eventId, isDraggable, timelineTabType, timelineId, isReadOnly }) => {
+  ({ browserFields, data, eventId, isDraggable, timelineTabType, scopeId, isReadOnly }) => {
     const containerElement = useRef<HTMLDivElement | null>(null);
     const dispatch = useDispatch();
-    const getTimeline = useMemo(() => timelineSelectors.getTimelineByIdSelector(), []);
+    const getScope = useMemo(() => {
+      if (isTimelineScope(scopeId)) {
+        return timelineSelectors.getTimelineByIdSelector();
+      } else if (isInTableScope(scopeId)) {
+        return dataTableSelectors.getTableByIdSelector();
+      }
+    }, [scopeId]);
+    const defaults = isTimelineScope(scopeId) ? timelineDefaults : tableDefaults;
```

Comment:
NIT: Maybe we could move this line inside the selector below? it is searching for the scope on every component update, but the `defaults` are almost never used, only inside the selector as a fallback.